src/jvm/main/org/jetbrains/kotlinx/lincheck/runner/FixedActiveThreadsExecutor.kt

```diff
@@ -30,11 +31,21 @@ internal class FixedActiveThreadsExecutor(private val nThreads: Int, runnerHash:
      */
     private val tasks = atomicArrayOfNulls<Any>(nThreads)
 
+    /**
+     * Spinners for each thread used for spin-wait on tasks.
+     */
+    private val taskSpinners = SpinnerList(nThreads)
+
     /**
      * null, waiting in [submitAndAwait] thread, DONE, or exception
      */
     private val results = atomicArrayOfNulls<Any>(nThreads)
 
+    /**
+     * Spinner for spin-wait on results.
```

Comment:
Please indicate that only the main thread that submits the tasks manipulates this spinner